<launch>
  <!-- Sunshine mission node params -->
  <arg name="robot_name" default="warpauv"/>

  <!-- GSCAM arguments -->
  <!-- <arg name="camera" default="/dev/video0" /> -->

  <!-- Vision arguments -->
  <arg name="scale" default="1.0" />
  <arg name="rate" default="5" />
  <arg name="hue" default="true" />
  <arg name="intensity" default="true" />
  <arg name="texton" default="true" />
  <arg name="orb" default="true" />
  <arg name="surf" default="false" />
  <arg name="image_topic" default="/usb_cam/image_raw" />

  <!-- Topic model arguments -->
  <arg name="K" default="10" />
  <arg name="V" default="16436" />
  <!--arg name="alpha" default="0.1" /-->
  <!--arg name="beta" default="1.0" /-->
  <!--arg name="gamma" default="1e-6" /-->
  <!--arg name="tau" default="0.5" /-->
  <arg name="alpha" default="0.073" />
  <arg name="beta" default="0.15" />
  <arg name="gamma" default="0" />
  <arg name="tau" default="0.5" />
  <arg name="p_refine_last_observation" default="0.5" />
  <arg name="num_threads" default="4" />
  <arg name="cell_space" default="32" />
  <arg name="G_time" default="1" />
  <arg name="G_space" default="1" />
  <arg name="polled_refine" default="false" />
  <arg name="words_topic" default="/rost_vision/words" />
  <arg name="min_obs_refine_time" default="200" />
  <arg name="word_obs_queue_size" default="1" />
  <arg name="publish_ppx" default="true" />
  <arg name="publish_local_surprise" default="true" />
  <arg name="publish_global_surprise" default="false" />

  <!-- Visualization arguments -->
  <!-- <arg name="words_topic" default="/rost_vision/words" />
  <arg name="topics_topic" default="/rost/topics" />
  <arg name="ppx_topic" default="/rost/cell_perplexity" />
  <arg name="show_topics" default="true" />
  <arg name="show_words" default="true" />
  <arg name="show_perplexity" default="true" />
  <arg name="show_equalized" default="true" /> -->
  


  <!-- Launch GSCAM with webcam -->
<!--   <env name="GSCAM_CONFIG" value="v4l2src device=$(arg camera) ! video/x-raw-rgb,width=640,height=480,framerate=30/1 ! videoconvert" /> -->
<!--   <node pkg="gscam" type="gscam" name="gscam"> -->
<!--   </node> -->
    <!-- <node pkg="usb_cam" type="usb_cam_node" name="usb_cam">
        <param name="pixel_format" value="yuyv" />
        <param name="camera_name" value="camera" />
        <param name="framerate" value="10" />
    </node> -->

  <!--<include file="$(find gscam)/examples/v4l.launch"></include>-->

  <!-- Launch word extractor -->
  <node pkg="sunshine" type="visual_word_extractor" name="rost_vision">
    <!--param name="vocabulary" type="string" value="$(find summarizer)/share/vocab/underwater1k.yml"/-->
    <!--<param name="vocabulary" type="string" value="$(find rost_vision)/data/orb_vocab/barbados2010_5000.yml"/>-->
    <param name="feature_descriptor" type="string" value="ORB"/>
    <param name="scale" type="double" value="$(arg scale)"/>
    <param name="rate" type="int" value="$(arg rate)"/>
    <param name="sensor_frame" type="string" value="map"/>
    <param name="use_hue" value="$(arg hue)"/>
    <param name="use_intensity" value="$(arg intensity)"/>
    <param name="use_orb" value="$(arg orb)"/>
    <param name="publish_2d_words" value="true"/>
    <param name="publish_3d_words" value="false"/>
    <param name="use_surf" value="$(arg surf)"/>
    <param name="use_texton" value="$(arg texton)" />
    <param name="image" value="$(arg image_topic)"/>
    <param name="use_pc" value="false" />
    <param name="use_tf" value="false" />
  </node>

  <!-- Launch topic model -->
  <node pkg="sunshine" type="topic_model_txy" name="rost">
    <param name="K" type="int" value="$(arg K)" />
    <param name="V" type="int" value="$(arg V)" />
    <param name="alpha" type="double" value="$(arg alpha)" />
    <param name="beta" type="double" value="$(arg beta)" />
    <param name="gamma" type="double" value="$(arg gamma)" />
    <param name="tau" type="double" value="$(arg tau)" />
    <param name="p_refine_last_observation" type="double" value="$(arg p_refine_last_observation)" />
    <param name="num_threads" type="int" value="$(arg num_threads)" />
    <param name="cell_space" type="int" value="$(arg cell_space)" />
    <param name="G_time" type="int" value="$(arg G_time)" />
    <param name="G_space" type="int" value="$(arg G_space)" />
    <param name="polled_refine" value="$(arg polled_refine)" />
    <param name="words_topic" value="$(arg words_topic)_2d" />
    <param name="min_obs_refine_time" value="$(arg min_obs_refine_time)" />
    <param name="word_obs_queue_size" value="$(arg word_obs_queue_size)" />
    <param name="publish_ppx" value="$(arg publish_ppx)" />
    <param name="publish_local_surprise" value="$(arg publish_local_surprise)" />
    <param name="publish_global_surprise" value="$(arg publish_global_surprise)" />
  </node>

  <!-- <node pkg="sunshine" type="visualize" name="visualizer" output="screen" >
    <param name="words_topic" type="string" value="$(arg words_topic)_2d" />
    <param name="topics_topic" type="string" value="$(arg topics_topic)" />
    <param name="ppx_topic" type="string" value="$(arg ppx_topic)" />
    <param name="show_topics" type="bool" value="$(arg show_topics)" />
    <param name="show_words" type="bool" value="$(arg show_words)" />
    <param name="show_perplexity" type="bool" value="$(arg show_perplexity)" />
    <param name="show_equalized" type="bool" value="$(arg show_equalized)" />
    <param name="image" value="$(arg image_topic)"/>
   </node> -->
</launch>
